core/primitives-core/src/account.rs

```diff
@@ -134,7 +171,7 @@ impl Account {
     pub fn code_hash(&self) -> CryptoHash {
```

Comment:
Replace `code_hash` with `contract` returning `AccountContractType`.  This way we ensure at compile time to properly handle all cases where `code_hash` was previously used.